server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -909,16 +920,39 @@ public boolean shouldUseConcurrentSearch() {
      * Evaluate if parsed request supports concurrent segment search
      */
     public void evaluateRequestShouldUseConcurrentSearch() {
```

Comment:
We can leave the logic same as before here and only add the condition to `evaluateAutoMode` if mode is `Auto`. The return value of that will be set to `requestShouldUseConcurrentSearch `. That way `shouldUseConcurrentSearch` will first check for the mode and if that is `Auto` or `All` then only it will check for `requestShouldUseConcurrentSearch`.